In [2]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2

from src.data import make_dataset

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [3]:
import pandas as pd

In [13]:
df = pd.read_excel("../data/raw/Base datos configurada INT (Actualizada 26-05-2023).xls")

## Cuantas glosas distintas hay en Diagnostico Principal?

In [17]:
conteo_raw = df['DIAGNOSTICO PRINCIPAL'].value_counts()
print(f"La base de datos original tiene {len(conteo_raw)} diagnosticos distintos \n")
print("Los diagnosticos y su conteo son:")
display(dict(conteo_raw))

La base de datos original tiene 335 diagnosticos distintos 

Los diagnosticos y su conteo son:


{'CIA': 441,
 'Tetralogía de Fallot': 328,
 'CIV': 199,
 'D-TGA': 179,
 'Ventriculo Unico Funcional': 113,
 'Estenosis Aortica': 101,
 'Anomalía Valvula Tricuspide': 85,
 'Estenosis Pulmonar': 80,
 'tetralogia de fallot': 68,
 'Canal AV': 63,
 'cia': 61,
 'DSVD': 61,
 'L-TGA': 59,
 'Coartación Aórtica': 55,
 'Miocardiopatía': 47,
 'Hipertensión Pulmonar   ': 43,
 'ventriculo unico funcional': 42,
 'Atresia Pulmonar': 38,
 'Atresia pulmonar': 34,
 'DAP': 34,
 'estenosis aortica': 33,
 'coartacion aortica': 33,
 'Valvulopatía no obstructiva': 32,
 'Electrofisiología': 24,
 'estenosis pulmonar': 20,
 'Otro': 18,
 'TGA': 18,
 'DTGA': 17,
 'CIA ': 16,
 'tronco arterioso': 16,
 'atresia pulmonar': 16,
 'Ventriculo unico funcional': 15,
 'Tetralogía de fallot': 15,
 'DVAPP': 15,
 'civ': 14,
 'anomalia valvula tricuspide': 14,
 'Estenosis aortica': 13,
 'Tetralogia de fallot': 12,
 'ventrículo unico funcional': 11,
 'anomalias de la valvula tricuspide': 11,
 'CIV ': 11,
 'FOP': 11,
 'estenosis

## Cuantas glosas distintas hay al tratar los datos?

In [18]:
df_procesada = pd.read_csv("../data/interim/datos_limpios.csv", encoding="latin-1", sep=";")

In [19]:
conteo_procesadas = df_procesada["DIAGNOSTICO PRINCIPAL"].value_counts()
print(f"La base de datos procesada tiene {len(conteo_procesadas)} diagnosticos distintos \n")
print("Los diagnosticos y su conteo son:")
display(dict(conteo_procesadas))

La base de datos procesada tiene 200 diagnosticos distintos 

Los diagnosticos y su conteo son:


{'cia': 519,
 'tetralogia de fallot': 449,
 'civ': 225,
 'ventriculo unico funcional': 209,
 'd-tga': 186,
 'estenosis aortica': 171,
 'coartacion aortica': 129,
 'estenosis pulmonar': 114,
 'atresia pulmonar': 102,
 'anomalia valvula tricuspide': 99,
 'canal av': 85,
 'dsvd': 62,
 'l-tga': 59,
 'hipertension pulmonar': 50,
 'miocardiopatia': 49,
 'dap': 34,
 'valvulopatia no obstructiva': 32,
 'tronco arterioso': 26,
 'electrofisiologia': 24,
 'otro': 20,
 'tga': 20,
 'dtga': 17,
 'dvapp': 15,
 'doble salida vd (dsvd)': 15,
 'fop': 12,
 'anomalias de la valvula tricuspide': 12,
 'anomalia de la valvula tricuspide': 10,
 'anomalias arterias coronarias': 10,
 'corazon normal': 9,
 'anomalias valvula tricuspide': 8,
 'tetralogia fallot': 8,
 'sindrome shine': 8,
 'drenaje venoso anomalo pulmonar parcial': 8,
 'interrupcion arco aortico': 8,
 'dvapt': 7,
 'estenosis valvular aortica': 7,
 'estenosis mitral': 6,
 'valvula aortica bicuspide': 6,
 'valvulopatias no obstructivas': 6,
 'civ + 

- Por lo tanto, se puede ver una reduccion de 335 a 200 diagnosticos distintos!

- La consolidacion de diagnosticos se puede seguir realizando si es que se utiliza
un algoritmo de agrupacion no supervisada (como K-Means)

In [6]:
vectorizador = CountVectorizer(stop_words=list(STOPWORDS_ESPANOL))

In [7]:
a = vectorizador.fit_transform(diags_procesados.unique())

In [60]:
modelo = KMeans(n_clusters=110, n_init=10, random_state=1)
modelo.fit(a)

KMeans(n_clusters=110, n_init=10, random_state=1)

In [61]:
clusters = pd.DataFrame({"diag": diags_procesados.unique(), "cluster": modelo.labels_})

In [62]:
clusters.sort_values("cluster").to_excel("prueba_cluster.xlsx", index=False)